core/src/main/scala/kafka/admin/ZkSecurityMigrator.scala

```diff
@@ -0,0 +1,247 @@
+              Future {
+                val childPathBuilder = new StringBuilder
+                for(child <- children.asScala) {
+                  childPathBuilder.clear
+                  childPathBuilder.append(path)
+                  childPathBuilder.append("/")
+                  childPathBuilder.append(child.mkString)
+
+                  val childPath = childPathBuilder.mkString
+                  setAclsRecursively(childPath)
+                }
+                promise success "done"
+              }(threadExecutionContext)
+          case Code.CONNECTIONLOSS =>
+            Future {
+              zkHandle.getChildren(path, false, GetChildrenCallback, ctx)
+            }(threadExecutionContext)
+          case Code.NONODE =>
+            warn("Node is gone, it could be have been legitimately deleted: %s".format(path))
+          case Code.SESSIONEXPIRED =>
+            // Starting a new session isn't really a problem, but it'd complicate
+            // the logic of the tool, so we quit and let the user re-run it.
+            error("ZooKeeper session expired while changing ACLs")
+            promise failure ZkException.create(KeeperException.create(Code.get(rc)))
+          case _ =>
+            error("Unexpected return code: %d".format(rc))
+            promise failure ZkException.create(KeeperException.create(Code.get(rc)))
+        }
+      }
+  }
+  
+  object SetACLCallback extends StatCallback {
+    def processResult(rc: Int,
+                      path: String,
+                      ctx: Object,
+                      stat: Stat) {
+      val zkHandle = zkUtils.zkConnection.getZookeeper
+      val promise = ctx.asInstanceOf[Promise[String]]
+      
+      Code.get(rc) match {
+          case Code.OK =>
+            info("Successfully set ACLs for %s".format(path))
+            promise success "done"
+          case Code.CONNECTIONLOSS =>
+            Future {
+              zkHandle.setACL(path, ZkUtils.DefaultAcls(zkUtils.isSecure), -1, SetACLCallback, ctx)
+            }(threadExecutionContext)
+          case Code.NONODE =>
+            warn("Node is gone, it could be have been legitimately deleted: %s".format(path))
+          case Code.SESSIONEXPIRED =>
+            // Starting a new session isn't really a problem, but it'd complicate
+            // the logic of the tool, so we quit and let the user re-run it.
+            error("ZooKeeper session expired while changing ACLs")
+            promise failure ZkException.create(KeeperException.create(Code.get(rc)))
+          case _ =>
+            error("Unexpected return code: %d".format(rc))
+            promise failure ZkException.create(KeeperException.create(Code.get(rc)))   
+      }
+    }
+  }
+
+  def run(args: Array[String]) {
+    var jaasFile = System.getProperty(JaasUtils.JAVA_LOGIN_CONFIG_PARAM)
+    val parser = new OptionParser()
+
+    val goOpt = parser.accepts("kafka.go", "Indicates whether to make this cluster secure or unsecure. The "
```

Comment:
Could we get a better name than kafka.go? How about zookeeper-acl? Also, the description should say this is to make the Kafka paths in ZK secure.